protos/hub/hub.proto

```diff
@@ -0,0 +1,132 @@
+syntax = "proto3";
+
+package hub;
+
+// gogo plugin toggles
+option (gogoproto.gogoproto_import) = true;
+option (gogoproto.goproto_registration) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.messagename_all) = true;
+option (gogoproto.sizer_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+// golang option
+option go_package = "hub";
+// java options
+option java_multiple_files = true;
+option java_outer_classname = "HubProto";
+option java_package = "io.btfs.hub";
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+import "google/protobuf/timestamp.proto";
+
+service Settings {
+  rpc GetSettings(SettingsReq) returns (SettingsResp);
+}
+
+service Hosts {
+  rpc GetHosts(HostsReq) returns (HostsResp);
+}
+
+message SettingsReq {
+  string id = 1;
+  int32 resp_size = 2;
+}
+
+enum ResponseCode {
+  SUCCESS = 0;
+  SIGNATURE_FORMAT_ERROR = 1;
+  COMPUTE_ADDRESS_ERROR = 2;
+  OTHER_ERROR = 20;
+}
+
+message SettingsResp {
+  ResponseCode code = 1;
+  string message = 2;
+  google.protobuf.Timestamp response_time = 3 [
+    (gogoproto.nullable) = false,
+    (gogoproto.stdtime) = true
+  ];
+  SettingsData settings_data = 4;
+}
+
+message SettingsData {
+  double storage_price_ask = 1;
+  double bandwidth_price_ask = 2;
+  double storage_time_min = 3;
+  double bandwidth_limit = 4;
+  double collateral_stake = 5;
+}
```

Comment:
concern with hub query, do not merge, yet

Reply:
got it